src/INode.java

```diff
@@ -204,12 +204,20 @@ private ASTValue evalCreateClass(WordsEnvironment environment) {
 		throw new AssertionError("Not yet implemented");	
 	}
 
-	private ASTValue evalCreateObj(WordsEnvironment environment) {
+	private ASTValue evalCreateObj(WordsEnvironment environment) throws WordsException {
 		ASTValue objName = children.get(0).eval(environment);
 		ASTValue className = children.get(1).eval(environment);
 		ASTValue properties = children.get(2) != null ? children.get(2).eval(environment) : null;
 		ASTValue position = children.get(3).eval(environment);
 
+		if (environment.getObject(objName.stringValue) != null) {
```

Comment:
Per my comment, I kind of like it if this line and the following stayed within the environment, and the environment handles the checks whether the creation is valid.  It separated the concerns better in case we change some things about how objects are managed.  eval just needs to call createObject, and the environment handles the rest.